core/http/server-spring/src/main/java/org/eclipse/rdf4j/http/server/repository/transaction/Transaction.java

```diff
@@ -134,13 +137,12 @@ void begin(IsolationLevel level)
 	void rollback()
 		throws InterruptedException, ExecutionException
 	{
-		Future<Boolean> result;
-		synchronized (futures) {
-			result = executor.submit(() -> {
-				txnConnection.rollback();
-				return true;
-			});
+		Future<Boolean> result = executor.submit(() -> {
+			txnConnection.rollback();
+			return true;
+		});
 
+		synchronized (futures) {
```

Comment:
Like I mentioned above, this needs to be rolled back I think. It is now possible that `hasActiveOperations` is called concurrently with this operation and returns `false`, despite the rollback already being scheduled for execution.